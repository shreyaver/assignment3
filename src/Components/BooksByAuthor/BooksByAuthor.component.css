.booksByAuthor {
    border: 5px solid orange;
    flex: 1;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    min-width: 310px;
    justify-content: space-evenly;
    width: 100%;
    border-radius: 30px;
}
.booksByAuthorContainer {
    margin-top: 50px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center; 
}
.authorName{
    background-color: white;
    color: orange;
    box-sizing: border-box;
    height: 3rem;
    display: flex;
    justify-content: center;
    align-items: center;
    margin-bottom: 5px;
    padding: 0.5rem;
    width: 25%;
    position: relative;
    bottom: -1.5rem;

}
.authorName span {
    flex-grow: 1;
    vertical-align: middle;
    text-align: center;
    font-size: 1.25rem;
    font-style: italic;
}
@media all and (max-width: 600px) {
   .authorName span {
       font-size: 1rem;
   }
   .authorName {
       margin-bottom: 10px;
   }
}