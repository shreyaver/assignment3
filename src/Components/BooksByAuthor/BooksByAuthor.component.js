import React from 'react';
import PropTypes from 'prop-types';
import Book from '../Book/Book.component';
import './BooksByAuthor.component.css';

const BooksByAuthor = (props) => {
  const booksByAuthor = [];
  const { author, books } = props;
  books.forEach((book, index) => {
    booksByAuthor.push(<Book book={book} key={book.Name} index={index} />);
  });
  return (
    <div className="booksByAuthorContainer">
      <div className="authorName">
        <span>{author}</span>
      </div>
      <div className="booksByAuthor">
        {booksByAuthor}
      </div>
    </div>
  );
};
BooksByAuthor.propTypes = {
  books: PropTypes.arrayOf(
    PropTypes.shape({
      Name: PropTypes.string.isRequired,
      rating: PropTypes.number.isRequired,
      Author: PropTypes.string.isRequired,
      id: PropTypes.number.isRequired,
    }),
  ),
  author: PropTypes.string.isRequired,
};
export default BooksByAuthor;
