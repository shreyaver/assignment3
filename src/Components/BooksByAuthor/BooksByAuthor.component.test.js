import React from 'react';
import renderer from 'react-test-renderer';
import BooksByAuthor from './BooksByAuthor.component';
import mockData from '../../mockData';

describe('BooksByAuthor', () => {
  it('renders without crashing', () => {
    const booksByAuthor = renderer.create(<BooksByAuthor author={Object.keys(mockData)[0]} books={mockData[Object.keys(mockData)[0]]} />).toJSON();
    expect(booksByAuthor).toMatchSnapshot();
  });
});
