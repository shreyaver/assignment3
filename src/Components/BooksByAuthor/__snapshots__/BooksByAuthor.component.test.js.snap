// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksByAuthor renders without crashing 1`] = `
<div
  className="booksByAuthorContainer"
>
  <div
    className="authorName"
  >
    <span>
      J K Rowling
    </span>
  </div>
  <div
    className="booksByAuthor"
  >
    <div
      className="book"
    >
      <div
        className="imagePanel"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <div
        className="bookDataEven"
      >
        <p
          className="bookName"
        >
          Harry Potter and the Deathly Hallows (Harry Potter, #7)
        </p>
        <span>
          4.62
          <button
            className="likedHeartIcon"
            type="button"
          />
        </span>
      </div>
    </div>
    <div
      className="book"
    >
      <div
        className="imagePanel"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <div
        className="bookDataOdd"
      >
        <p
          className="bookName"
        >
          Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)
        </p>
        <span>
          4.54
          <button
            className="likedHeartIcon"
            type="button"
          />
        </span>
      </div>
    </div>
    <div
      className="book"
    >
      <div
        className="imagePanel"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <div
        className="bookDataEven"
      >
        <p
          className="bookName"
        >
          Harry Potter and the Half-Blood Prince (Harry Potter, #6)
        </p>
        <span>
          4.54
          <button
            className="likedHeartIcon"
            type="button"
          />
        </span>
      </div>
    </div>
    <div
      className="book"
    >
      <div
        className="imagePanel"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <div
        className="bookDataOdd"
      >
        <p
          className="bookName"
        >
          Harry Potter and the Goblet of Fire (Harry Potter, #4)
        </p>
        <span>
          4.53
          <button
            className="likedHeartIcon"
            type="button"
          />
        </span>
      </div>
    </div>
    <div
      className="book"
    >
      <div
        className="imagePanel"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <div
        className="bookDataEven"
      >
        <p
          className="bookName"
        >
          Harry Potter and the Order of the Phoenix (Harry Potter, #5)
        </p>
        <span>
          4.47
          <button
            className="heartIcon"
            type="button"
          />
        </span>
      </div>
    </div>
    <div
      className="book"
    >
      <div
        className="imagePanel"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <div
        className="bookDataOdd"
      >
        <p
          className="bookName"
        >
          Harry Potter and the Sorcerers Stone (Harry Potter, #1)
        </p>
        <span>
          4.45
          <button
            className="heartIcon"
            type="button"
          />
        </span>
      </div>
    </div>
    <div
      className="book"
    >
      <div
        className="imagePanel"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <div
        className="bookDataEven"
      >
        <p
          className="bookName"
        >
          Harry Potter and the Chamber of Secrets (Harry Potter, #2)
        </p>
        <span>
          4.38
          <button
            className="heartIcon"
            type="button"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
